Feat/raycasting basic rendering

mikkegt opened the pull request:
<img width="807" height="636" alt="image" src="https://github.com/user-attachments/assets/cf33b627-c9d5-4289-9d68-bb25fe8a3dfe" />

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Implemented a 3D rendering engine that displays walls, ceilings, and floors from the player's perspective
  * Added player position initialization and world environment setup

* **Documentation**
  * Added detailed technical documentation for the rendering system and core algorithm

* **Refactor**
  * Reorganized game initialization workflow
  * Updated build system configuration for new engine components

* **Style**
  * Improved code formatting and consistency

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
docs: add raycasting module design and structure

- Add raycasting algorithm documentation (DDA)
- Add raycasting module design and interface
- Add player and image structures to cub3d.h
- Add raycasting constants (MAP_WIDTH, MAP_HEIGHT, etc)
- Create raycasting.h header with t_ray structure
- Create init.c with init_hardcoded_map() and init_player()

Commit:
format

Commit:
refactor: add static prototype for init_mlx function

- Add static function prototype declaration at file top
- Make init_mlx static (internal function)
- Improve code organization following partner's style

Commit:
feat: implement basic raycasting rendering (WIP)

- Add DDA raycasting algorithm
- Initialize 8x8 hardcoded test map
- Set up player position and camera plane
- Render ceiling, walls, and floor
- Split functions to comply with 42 Norm (5 func/file)

WIP: Hardcoded map will be replaced with parser data during integration

mikkegt committed:
rename mock program

mikkegt committed:
update ray docs

nytsdayo commented on docs/examples/raycasting/raycast_demo:
exeファイルかな？バイナリが入っちゃってます

nytsdayo commented on srcs/engine/init/init_mlx.c:
```diff
@@ -6,29 +6,59 @@
+		write(2, "Error\nWindow creation failed\n", 29);
+		exit(EXIT_FAILURE);
+	}
 }
```
* filenameが`init_mlx.c`で、ソースファイルから外部に参照できる関数がinit_game()になっています！

Memo : ざっくりコーディングルールを明文化したいですね

nytsdayo reviewed (commented):
一旦、コードだけ見ました！！
どちらも極めて軽微ですが、2点だけ気になるところありました。
内部動作はまだ確認できていませんT_T

mikkegt committed:
refactor: rename init_mlx.c to init_game.c for consistency

- Rename init_mlx.c to init_game.c to match the main function name init_game()
- Update Makefile to reference the new filename
- Update header comment in init_game.c

This change improves code clarity by aligning the filename with its primary function.

mikkegt committed:
chore: remove executable files from repository

- Remove raycast_demo executable from docs/examples/raycasting/
- Remove prototype.o object file

Executable and object files should not be tracked in version control.
